api/datasets.go

```diff
@@ -306,6 +306,10 @@ func (h *DatasetHandlers) getHandler(w http.ResponseWriter, r *http.Request) {
 	}
 	res := lib.GetResult{}
 	err := h.Get(&p, &res)
+	if err == repo.ErrNoHistory {
```

Comment:
These no history errors should be handled inside the `if err != nil` check imo. Considering how verbose error handling is in go, I think it's worth keeping all error-returns within the `!= nil` branch